bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java

```diff
@@ -14,24 +14,30 @@
 
 import java.util.List;
 import java.util.Map;
+import java.util.TreeMap;
 
 import javax.script.ScriptEngine;
 
 /**
  * This class is used by the ScriptManager to load ScriptEngines.
  * This is meant as a way to allow other OSGi bundles to provide custom Script-Languages with special needs (like
- * Nashorn, Groovy, etc.)
+ * Nashorn, Jython, Groovy, etc.)
  *
  * @author Simon Merschjohann
- *
+ * @author Scott Rushworth - added getLanguageNames() for use in ScriptModuleTypeProvider
  */
 public interface ScriptEngineFactory {
 
     /**
-     * @return the list of supported language endings e.g. py, jy
+     * @return the list of supported language endings, e.g. py, groovy, js
      */
     List<String> getLanguages();
 
+    /**
+     * @return a map of supported language name : file extension, e.g. {Python=py, Groovy=groovy, ECMAScript=js}
+     */
+    TreeMap<String, String> getLanguageNames();
```

Comment:
Hm, I don't think "get language names" should return more that a language name.
The language name would be Python, Groovy, ECMAScript (WRT your comment).
Perhaps this is more about a language file extendsion mapping?

It is very uncommon to use the specific implementation of a map in an interface.
It should be enough to state a map will be returned. Which one should not matter.

Reply:
Please fix your JavaDoc comments.
A comment should start with a short description, empty line, param(s), return and throws statements.

(I comment this only once, it fits for the other methods, too)

Reply:
I had a hard time trying to decide on a method name. getLanguages wasn't very descriptove, but I chose getLanguageNames to differentiate between the two. I agree... both of them are not clear what they do. How about scriptEngineIdentifiers and scriptEngineNameMap?

I will clean up the TreeMaps. Java is new to me, so I'm going to make some beginner mistakes. Thank you for the direction.

> Please fix your JavaDoc comments.

I'll look at them again. I was mainly following the pattern of what was already in the file.